src/python/dxpy/bindings/apollo/dataset.py

```diff
@@ -0,0 +1,56 @@
+from dxpy import DXHTTPRequest
+
+
+class Dataset:
+    def __init__(self, record_id, project):
+        self.record_id = record_id
+        self.project = project
+
+        self.populate_http_request()
```

Comment:
It is not a good practice to trigger external resources or any heavier computation in the constructor. One of the reasons is to keep our ability to instantiate a new object without necessarily depending on the underlying environment.

Instead I would recommend lazy-loading - check for the request response object only when someone requests info from it:
```
def get_vizualise_info(self):
   if self.vizualise_info is None:
      self.vizualise_info = DXHTTPRequest(
            "/" + self.record_id + "/visualize",
            {"project": self.project, "cohortBrowser": False},
        )
  return self.vizualise_info
```
And then:
```
def get_record_name(self):
   return self.get_vizualise_info().get("recordName")
```